sklearn/model_selection/_split.py

```diff
@@ -287,9 +287,17 @@ def __init__(self, n_splits, shuffle, random_state):
             raise TypeError("shuffle must be True or False;"
                             " got {0}".format(shuffle))
 
+        if not shuffle and random_state is not None:  # None is the default
+            raise ValueError(
```

Comment:
Not sure if this should be an error.

Reply:
yeah, with the other comment, now I see why this is very useful.

But this needs a whats_new entry of its own.

Reply:
you know what maybe I'll make a separate PR that we can quickly review. After all this is unrelated to the proposed changes.